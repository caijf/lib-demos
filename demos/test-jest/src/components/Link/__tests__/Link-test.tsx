import React from 'react';
import { create, act, ReactTestRenderer, ReactTestRendererJSON } from 'react-test-renderer';
import Link from '../index';

test('Link changes the class when hovered', () => {
  let component: ReactTestRenderer | undefined;

  act(() => {
    component = create(<Link page="https://www.baidu.com/">百度</Link>);
  });

  expect(component?.toJSON()).toMatchSnapshot();

  // manually trigger the callback
  act(() => {
    (component?.toJSON() as ReactTestRendererJSON).props.onMouseEnter();
  });
  // re-rendering
  expect(component?.toJSON()).toMatchSnapshot();

  // manually trigger the callback
  act(() => {
    (component?.toJSON() as ReactTestRendererJSON).props.onMouseLeave();
  });
  // re-rendering
  expect(component?.toJSON()).toMatchSnapshot();
});
