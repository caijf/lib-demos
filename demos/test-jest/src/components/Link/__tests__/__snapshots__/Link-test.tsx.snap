// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://www.baidu.com/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  百度
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://www.baidu.com/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  百度
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://www.baidu.com/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  百度
</a>
`;
